### PR TITLE
Allow printf-style formatting in CkAbort/CmiAbort

### PR DESCRIPTION
**Pull request by matthiasdiener**

### CODE REVIEW

File: src/libs/ck-libs/datatransfer/charm_standalone.C [hunk 1]
```diff
@@ -5,8 +5,11 @@
 #include <stdlib.h>
 #include <stdarg.h> /*<- for va_start & friends */
 
-extern "C" void CmiAbort(const char *why) {
-	fprintf(stderr,"Fatal error> %s\n",why);
+extern "C" void CmiAbort(const char *fmt, ...) {
+	fprintf(stderr,"Fatal error>");
+    va_list p; va_start(p, fmt);
```
Comment (evan-charmworks) [inline src/libs/ck-libs/datatransfer/charm_standalone.C]: whitespace

Comment (matthiasdiener) [reply src/libs/ck-libs/datatransfer/charm_standalone.C]: fixed.

**Comment by matthiasdiener** [comment]
> We should tag CmiAbort (and probably also CmiPrintf and CmiError) with `__attribute__ ((format (printf, 1, 2)))` on GCC and clang so that it is checked for -Wformat-security warnings.
> 
> https://gcc.gnu.org/onlinedocs/gcc/Common-Function-Attributes.html#Common-Function-Attributes
> https://clang.llvm.org/docs/AttributeReference.html#format

Agreed, but this seems better in a separate PR?

**Comment by evan-charmworks** [comment]
> Agreed, but this seems better in a separate PR?

Potentially. There might be instances where the string passed to CmiAbort is runtime generated, which was previously perfectly valid, but now that the first argument is passed as a printf format string, could in theory have issues with % characters. This may not be an issue in practice.

**Comment by matthiasdiener** [comment]
> > Agreed, but this seems better in a separate PR?
> 
> Potentially. There might be instances where the string passed to CmiAbort is runtime generated, which was previously perfectly valid, but now that the first argument is passed as a printf format string, could in theory have issues with % characters. This may not be an issue in practice.

Yeah, that's a good point. I changed '%' to '%%' in our source. This might be worth mentioning in the release notes.

**Comment by evan-charmworks** [comment]
In that case, maybe we should add a new abort call that supports printf arguments instead of redefining what the current one is.

**Comment by matthiasdiener** [comment]
> In that case, maybe we should add a new abort call that supports printf arguments instead of redefining what the current one is.

I think we don't need a new call. Having a '%' in the string will only print a slightly incorrect message and should be easy to fix in the application (clang at least will also print a warning).

**Comment by evan-charmworks** [comment]
Something else worth discussing is whether the abort calls should add a trailing newline. Our own tests and examples and even NAMD are not consistent about whether the strings passed in have one or not.

**Comment by stwhite91** [comment]
IMO CmiAbort should add a newline to the string passed to it since having an extra newline is better than not having one at all in an abort statement

**Comment by matthiasdiener** [comment]
> Something else worth discussing is whether the abort calls should add a trailing newline. Our own tests and examples and even NAMD are not consistent about whether the strings passed in have one or not.

Updated the patch with a newline.

**Pull request merged**